dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java

```diff
@@ -163,120 +164,90 @@ public String answer(InvocationOnMock invocation) throws Throwable {
+        vanityUrl1 = filtersUtil.createVanityUrl("test link1", Host.SYSTEM_HOST, "/" + uniqueUrl + "1",
+                "/about-us/" + CMSFilter.CMS_INDEX_PAGE, 200, 1, defaultLanguageId);
+        filtersUtil.publishVanityUrl(vanityUrl1);
 
-            Logger.info(this.getClass(),
-                    "/testLink3 should redirect to http://demo.dotcms.com/about-us/"
-                            + CMSFilter.CMS_INDEX_PAGE);
-            request = getMockRequest(site.getHostname(), "/testLink3");
-            response = new MockResponseWrapper(Mockito.mock(HttpServletResponse.class));
-            filter.doFilter(request, response, chain);
-            Assert.assertEquals(301, response.getStatus());
-            Assert.assertEquals("http://demo.dotcms.com/about-us/" + CMSFilter.CMS_INDEX_PAGE,
-                    response.getRedirectLocation());
+        vanityUrl2 = filtersUtil.createVanityUrl("test link2", site.getIdentifier(),
+                "/" + uniqueUrl + "2", "/about-us/" + CMSFilter.CMS_INDEX_PAGE, 200, 1,
+                defaultLanguageId);
+        filtersUtil.publishVanityUrl(vanityUrl2);
 
-            Logger.info(this.getClass(),
-                    "/testLink4 should redirect to http://demo.dotcms.com/about-us/"
-                            + CMSFilter.CMS_INDEX_PAGE);
-            request = getMockRequest(site.getHostname(), "/testLink4");
-            response = new MockResponseWrapper(Mockito.mock(HttpServletResponse.class));
-            filter.doFilter(request, response, chain);
-            Assert.assertEquals(301, response.getStatus());
-            Assert.assertEquals("http://demo.dotcms.com/about-us/" + CMSFilter.CMS_INDEX_PAGE,
-                    response.getRedirectLocation());
+        vanityUrl3 = filtersUtil.createVanityUrl("test link3", site.getIdentifier(),
+                "/" + uniqueUrl + "3", "http://demo.dotcms.com/about-us/" + CMSFilter.CMS_INDEX_PAGE,
+                301, 1, defaultLanguageId);
+        filtersUtil.publishVanityUrl(vanityUrl3);
 
-            Logger.info(this.getClass(),
-                    "/forbidden should forward to /products/" + CMSFilter.CMS_INDEX_PAGE);
-            request = getMockRequest(site.getHostname(), "/forbidden");
-            response = new MockResponseWrapper(Mockito.mock(HttpServletResponse.class));
-            filter.doFilter(request, response, chain);
-            Assert.assertEquals(200, response.getStatus());
+        vanityUrl4 = filtersUtil.createVanityUrl("test link4", site.getIdentifier(),
+                "/" + uniqueUrl + "4", "http://demo.dotcms.com/about-us/" + CMSFilter.CMS_INDEX_PAGE,
+                301, 1, defaultLanguageId);
+        filtersUtil.publishVanityUrl(vanityUrl4);
 
-        } catch (Exception e) {
+        vanityUrl5 = filtersUtil.createVanityUrl("test link5", Host.SYSTEM_HOST,
+                "/forbidden", "/products/" + CMSFilter.CMS_INDEX_PAGE, 302, 1, defaultLanguageId);
```

Comment:
Codacy found an issue: [The String literal "/products/" appears 4 times in this file; the first occurrence is on line 198](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)